fix: update `math/base/special/cosd` to match correct reference implementation

anandkaranubc opened the pull request:
Resolves none.

## Description

> What is the purpose of this pull request?

This pull request:

- Fixes the implementation of `math/base/special/cosd` to align with the correct reference implementation.
- Adds improved test fixtures to enhance accuracy and ensure correctness.

## Related Issues

> Does this pull request have any related issues?

This pull request:

- resolves none

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

anandkaranubc committed:
test: add improved test fixtures

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
fix: update the function according to reference implementation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

anandkaranubc commented:
Reference implementation: https://github.com/JuliaLang/julia/blob/9ef2af9db620644253941e080b5b4986ad0d6091/base/special/trig.jl#L1048-L1072

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/cosd/src/main.c:
```diff
@@ -34,12 +37,30 @@
+
+	if ( rx <= 45.0 ) {
+		xRad = stdlib_base_deg2rad( rx );
+		return stdlib_base_kernel_cos( xRad, 0.0 );
```
```suggestion
		return stdlib_base_kernel_cos( stdlib_base_deg2rad( rx ), 0.0 );
```
@anandkaranubc Can we directly do something like this, eliminating the need for `xRad`?

anandkaranubc replied:
Sure! Doing it right now. The only reason I did it that way was to improve code readability and keep the code structured.

kgryte replied:
I would say we typically try to avoid temporary variables which can be readily inlined, so long as the expressions are relatively simple. In this case, Gunj's suggestion is generally appropriate.

anandkaranubc replied:
Well noted!  

Thanks, @gunjjoshi, for the suggestion. 🙌

anandkaranubc committed:
Merge branch 'stdlib-js:develop' into fix/cosd

anandkaranubc committed:
chore: remove xRad for cleaner code

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/cosd/test/fixtures/julia/runner.jl:
```diff
@@ -19,7 +19,7 @@
 
 """
-    gen( domain, name )
+	gen( domain, name )
```
```suggestion
    gen( domain, name )
```

gunjjoshi replied:
We should use spaces here.

anandkaranubc replied:
Thanks! 😄

anandkaranubc replied:
The linter complains, by the way, if I make this change locally, with the message: `Wrong indentation type (spaces instead of tabs).`

gunjjoshi replied:
@kgryte Using spaces here, instead of a tab (as used in packages such as `exp`: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/exp/test/fixtures/julia/runner.jl#L22) leads to linting errors. Is there a need to suppress the lint error, or, tabs are to be used here, while updating the same in `exp` too?

gunjjoshi replied:
@kgryte We've got something similar in packages such as [`pow`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/pow/test/fixtures/julia/runner.jl#L22), [`erf`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/erf/test/fixtures/julia/runner.jl#L23), [`sqrt`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/sqrt/test/fixtures/julia/runner.jl#L22), etc.

kgryte replied:
Sigh. This is where EditorConfig linting is a bit lacking.

kgryte replied:
It is fine to use tabs. We don't have much of a choice due to the EditorConfig Checker not distinguishing between code and docs.

gunjjoshi replied:
Alright, thanks!

kgryte replied:
It is also not clear why Julia requires that the function signature be indented (see https://docs.julialang.org/en/v1/manual/documentation/), but oh well.

gunjjoshi committed:
Update lib/node_modules/@stdlib/math/base/special/cosd/test/fixtures/julia/runner.jl

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

gunjjoshi committed:
style: use tab in runner.jl

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

Planeshifter reviewed (approved):
Looks all good to me, as well. Thanks @anandkaranubc and @gunjjoshi!
